featurebyte/service/observation_table.py

```diff
@@ -250,11 +264,62 @@ def __init__(  # pylint: disable=too-many-arguments
         self.temp_storage = temp_storage
         self.primary_entity_validator = primary_entity_validator
         self.use_case_service = use_case_service
+        self.target_namespace_service = target_namespace_service
+        self.target_service = target_service
 
     @property
     def class_name(self) -> str:
         return "ObservationTable"
 
+    async def _validate_columns(
+        self,
+        available_columns: List[str],
+        primary_entity_ids: Optional[List[PydanticObjectId]],
+        target_column: Optional[str],
+    ) -> Optional[ObjectId]:
+        # Check if required column names are provided
+        missing_columns = []
+
+        # Check if point in time column is present
+        if SpecialColumnName.POINT_IN_TIME not in available_columns:
+            missing_columns.append(str(SpecialColumnName.POINT_IN_TIME))
+
+        # Check if entity columns are present
+        if primary_entity_ids:
+            for entity_id in primary_entity_ids:
+                entity = await self.entity_service.get_document(document_id=entity_id)
+                if not set(entity.serving_names).intersection(available_columns):
+                    missing_columns.append("/".join(entity.serving_names))
+
+        # Check if target column is present
+        if target_column:
+            if target_column not in available_columns:
+                missing_columns.append(target_column)
+            target_namespaces = await self.target_namespace_service.list_documents_as_dict(
+                query_filter={"name": target_column}
+            )
+            if target_namespaces["total"] == 0:
+                raise ObservationTableInvalidTargetNameError(
+                    f"Target name not found: {target_column}"
+                )
+
+            # validate target namespace has same primary entity ids
+            target_namespace = target_namespaces["data"][0]
+            if target_namespace["entity_ids"] != primary_entity_ids:
```

Comment:
Are both the entity_ids sorted already?

Reply:
`primary_entity_ids` is sorted. Need to confirm for `target_namespace["entity_ids"]`

Reply:
Checked and confirmed that `entity_ids` are sorted by the pydantic validator in:
https://github.com/featurebyte/featurebyte/blob/main/featurebyte/models/feature_namespace.py#L126